src/simplewallet/simplewallet.cpp

```diff
@@ -7046,62 +7061,50 @@ bool simple_wallet::ons_by_owner(const std::vector<std::string>& args)
       }
 
-      if (requests.back().entries.size() >= cryptonote::rpc::ONS_OWNERS_TO_NAMES::MAX_REQUEST_ENTRIES)
-        requests.emplace_back();
-      requests.back().entries.push_back(arg);
+      req_params["entries"].push_back(arg);
     }
   }
 
-  rpc_results.reserve(requests.size());
-  for (auto const &request : requests)
+  auto [success, result] = m_wallet->ons_owners_to_names(req_params);
+  if (!success)
   {
-    auto [success, result] = m_wallet->ons_owners_to_names(request);
-    if (!success)
-    {
-      fail_msg_writer() << "Connection to daemon failed when requesting ONS names";
-      return false;
-    }
-    rpc_results.emplace_back(std::move(result));
+    fail_msg_writer() << "Connection to daemon failed when requesting ONS names";
+    return false;
   }
 
-
   auto nettype = m_wallet->nettype();
-  for (size_t i = 0; i < rpc_results.size(); i++)
+  for (auto const &entry : result["entries"])
   {
-    auto const &rpc = rpc_results[i];
-    for (auto const &entry : rpc)
+    std::string_view name;
+    std::string value;
+    if (auto got = cache.find(entry["name_hash"]); got != cache.end())
     {
-      std::string_view name;
-      std::string value;
-      if (auto got = cache.find(entry.name_hash); got != cache.end())
-      {
-        name = got->second.name;
-        ons::mapping_value mv;
-        if (ons::mapping_value::validate_encrypted(entry.type, oxenmq::from_hex(entry.encrypted_value), &mv)
-            && mv.decrypt(name, entry.type))
-          value = mv.to_readable_value(nettype, entry.type);
-      }
-
-      auto writer = tools::msg_writer();
-      writer
-        << "Name (hashed): " << entry.name_hash;
-      if (!name.empty()) writer
-        << "\n    Name: " << name;
-      writer
-        << "\n    Type: " << entry.type;
-      if (!value.empty()) writer
-        << "\n    Value: " << value;
-      writer
-        << "\n    Owner: " << entry.owner;
-      if (entry.backup_owner) writer
-        << "\n    Backup owner: " << *entry.backup_owner;
-      writer
-        << "\n    Last updated height: " << entry.update_height;
-      if (entry.expiration_height) writer
-        << "\n    Expiration height: " << *entry.expiration_height;
-      writer
-        << "\n    Encrypted value: " << entry.encrypted_value;
+      name = got->second.name;
+      ons::mapping_value mv;
+      if (ons::mapping_value::validate_encrypted(static_cast<ons::mapping_type>(entry["type"].get<uint16_t>()), oxenmq::from_hex(entry["encrypted_value"].get<std::string>()), &mv)
+          && mv.decrypt(name, static_cast<ons::mapping_type>(entry["type"].get<uint16_t>())))
+        value = mv.to_readable_value(nettype, static_cast<ons::mapping_type>(entry["type"].get<uint16_t>()));
     }
+
+    auto writer = tools::msg_writer();
+    writer
+      << "Name (hashed): " << entry["name_hash"];
+    if (!name.empty()) writer
+      << "\n    Name: " << name;
+    writer
+      << "\n    Type: " << entry["type"];
```

Comment:
i dont think this will print the mapping type anymore as its no longer got the ons_type to string function now